move/walrus_site/sources/site.move

```diff
@@ -53,13 +65,40 @@ module walrus_site::site {
     public fun new_resource(
         path: String,
         blob_id: u256,
-        blob_hash: u256
+        blob_hash: u256,
+        range_start: Option<u256>,
+        range_end: Option<u256>
     ): Resource {
+        let start_is_defined = option::is_some(&range_start);
+        let end_is_defined = option::is_some(&range_end);
+        // If defined, upper range bound should be positive (zero excluded).
+        if (end_is_defined) {
+            let end = option::borrow(&range_end);
+            assert!( *end > 0, ERangeEndNotPositive);
+        };
```

Comment:
Remark: `u256` cannot be negative.

Question: is this to avoid ranges of length 0?

Reply:
Yes it is done for checking zero lengths!